### ISSUE TITLE
🐛 Bug: README.md badges don't have sufficient white-on-green color contrast

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [JoshuaKGoldberg/create-typescript-app#1224]
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

https://webaim.org/articles/contrast: all text in the README.md should have >= 3.5:1 color contrast between foreground and background text.

### Actual

The badges I've been using -which, I should add, use pretty a common default green 😬- only have 2.68:1: https://webaim.org/resources/contrastchecker/?fcolor=FFFFFF&bcolor=21BB42. 

### Additional Info

The images do apply a shadow of `#218334`:

![Screenshot of the '💪 typescript: strict' README.md badge, zoomed in, with white text, a #218334 shadow, and #31B548 background](https://github.com/JoshuaKGoldberg/create-typescript-app/assets/3335181/52aceb0b-dd42-4db7-9c0a-f1daa8cab002)

...but that's only underneath the text. So my hunch is it really only "counts" halfway.

**Comment by JoshuaKGoldberg** [comment]
I like `#289835`:

![Screenshot of six README badges with white text on #289835 green background, except for 'coverage: 95%' with a lime green background](https://github.com/JoshuaKGoldberg/create-typescript-app/assets/3335181/9bff2354-96e6-4a67-be77-c51552951d2c)

...but the codecov badge will need to updated with a custom range of colors, I think. Blocked on https://github.com/badges/shields/issues/994 -> https://github.com/badges/shields/issues/9890.

**Comment by JoshuaKGoldberg** [comment]
Aha, the shields work was merged! So this is ✅. Yay!!!
